File: examples/python/create_video.py
import json
import os
import urllib.request


API_KEY = os.environ["USVIDEOAPI_KEY"]
BASE_URL = os.environ.get("USVIDEOAPI_BASE_URL", "https://usvideoapi.com")

payload = {
    "model": "seedance-1-pro",
    "prompt": (
        "A 9:16 local restaurant ad showing a lunch combo, close-up food shots, "
        "quick pacing, and a final offer CTA."
    ),
    "size": "1080p",
    "duration": 5,
}

req = urllib.request.Request(
    f"{BASE_URL}/v1/videos",
    data=json.dumps(payload).encode("utf-8"),
    headers={
        "Content-Type": "application/json",
        "Authorization": f"Bearer {API_KEY}",
    },
    method="POST",
)

with urllib.request.urlopen(req) as resp:
    print(resp.read().decode("utf-8"))

File: examples/javascript/create-video.mjs
const apiKey = process.env.USVIDEOAPI_KEY;
const baseUrl = process.env.USVIDEOAPI_BASE_URL || "https://usvideoapi.com";

if (!apiKey) {
  throw new Error("Set USVIDEOAPI_KEY first.");
}

const payload = {
  model: "seedance-2.0",
  prompt:
    "A 9:16 ad for a dental clinic with clean office visuals, smiling patient moments, trust-building text overlays, and a booking CTA.",
  resolution: "1080p",
  duration: 5,
};

const response = await fetch(`${baseUrl}/v1/videos`, {
  method: "POST",
  headers: {
    "Content-Type": "application/json",
    Authorization: `Bearer ${apiKey}`,
  },
  body: JSON.stringify(payload),
});

console.log(await response.text());
